pkg/ruler/config.go

```diff
@@ -48,14 +48,25 @@ func (c *Config) Validate() error {
 }
 
 type RemoteWriteConfig struct {
-	Client              config.RemoteWriteConfig `yaml:"client"`
-	Enabled             bool                     `yaml:"enabled"`
-	ConfigRefreshPeriod time.Duration            `yaml:"config_refresh_period"`
+	Client              *config.RemoteWriteConfig  `yaml:"client,omitempty"`
+	Clients             []config.RemoteWriteConfig `yaml:"clients"`
```

Comment:
This looks like adding a slice of remote write configs for a single tenant. Should we actually have a separate config per tenant? Shouldn't this be a `map[string]config.RemoteWriteConfig`. I suggest you take as good example of how to introduce a config per tenant the following in per-tenant-ruler-configs `ruler_remote_write_*` in the `limits_config` section:  https://grafana.com/docs/loki/latest/configuration/#limits_config

Reply:
it's tricky because, if I understand correctly, multi tenancy for the ruler is handled by `limit_config` overrides. By using a map, are you suggesting that we handle multi tenancy for ruler directly in the configs instead of overrides?